pkg/reconciler/metric/metric_test.go

```diff
@@ -159,114 +158,43 @@ func TestReconcile(t *testing.T) {
-		createOrUpdateCalls: true,
-		expectErr:           updateError,
-	}, {
-		name: "error on delete",
-		key:  "old/metric",
-		action: func(scs servingclientset.Interface) {
-			m := metric("new", "metric")
-			scs.AutoscalingV1alpha1().Metrics(m.Namespace).Create(m)
-			scs.AutoscalingV1alpha1().Metrics(m.Namespace).Delete(m.Name, &metav1.DeleteOptions{})
-		},
-		collector:   &testCollector{deleteError: deleteError},
-		deleteCalls: true,
-		expectErr:   deleteError,
-	}}
-
-	for _, tt := range tests {
-		t.Run(tt.name, func(t *testing.T) {
-			ctx, cancel, informers := SetupFakeContextWithCancel(t)
-
-			tt.collector.createOrUpdateCalls = make(chan struct{}, 100)
-			tt.collector.deleteCalls = make(chan struct{}, 100)
+	collector := &testCollector{}
+	collector.createOrUpdateCalls = make(chan struct{}, 100)
+	collector.deleteCalls = make(chan struct{}, 100)
 
-			scs := servingclient.Get(ctx)
+	ctl := NewController(ctx, configmap.NewStaticWatcher(), collector)
 
-			ctl := NewController(ctx, configmap.NewStaticWatcher(), tt.collector)
-
-			wf, err := controller.RunInformers(ctx.Done(), informers...)
-			if err != nil {
-				cancel()
-				t.Fatalf("StartInformers() = %v", err)
-			}
-
-			var eg errgroup.Group
-			eg.Go(func() error { return ctl.Run(1, ctx.Done()) })
-			defer func() {
-				cancel()
-				wf()
-				eg.Wait()
-			}()
-
-			tt.action(scs)
+	wf, err := controller.RunInformers(ctx.Done(), informers...)
+	if err != nil {
+		cancel()
+		t.Fatalf("StartInformers() = %v", err)
+	}
 
-			if !tt.createOrUpdateCalls {
-				select {
-				case <-tt.collector.createOrUpdateCalls:
-					t.Error("CreateOrUpdate() called non-zero times, want 0 times")
-				case <-time.After(1 * time.Second):
-				}
-			} else {
-				select {
-				case <-tt.collector.createOrUpdateCalls:
-				case <-time.After(1 * time.Second):
-					t.Error("CreateOrUpdate() called 0 times, want non-zero times")
-				}
-			}
+	var eg errgroup.Group
+	eg.Go(func() error { return ctl.Run(1, ctx.Done()) })
+	defer func() {
+		cancel()
+		wf()
+		eg.Wait()
+	}()
+
+	m := metric("new", "metric")
+	scs := servingclient.Get(ctx)
+
+	scs.AutoscalingV1alpha1().Metrics(m.Namespace).Create(m)
+	select {
+	case <-collector.createOrUpdateCalls:
+	case <-time.After(1 * time.Second):
+		t.Error("CreateOrUpdate() called 0 times, want non-zero times")
+	}
 
-			if !tt.deleteCalls {
-				select {
-				case <-tt.collector.deleteCalls:
-					t.Error("CreateOrUpdate() called non-zero times, want 0 times")
-				case <-time.After(1 * time.Second):
-				}
-			} else {
-				select {
-				case <-tt.collector.deleteCalls:
-				case <-time.After(1 * time.Second):
-					t.Error("CreateOrUpdate() called 0 times, want non-zero times")
-				}
-			}
-		})
+	scs.AutoscalingV1alpha1().Metrics(m.Namespace).Delete(m.Name, &metav1.DeleteOptions{})
+	select {
+	case <-collector.deleteCalls:
+	case <-time.After(1 * time.Second):
+		t.Error("CreateOrUpdate() called 0 times, want non-zero times")
```

Comment:
```suggestion
		t.Error("Delete() called 0 times, want non-zero times")
```

I think this was my bug 😬